x-pack/plugins/alerts/server/alerts_client.ts

```diff
@@ -396,6 +396,17 @@ export class AlertsClient {
     };
   }
 
+  public async hasDecryptionFailures(): Promise<boolean> {
+    const { saved_objects: data } = await this.unsecuredSavedObjectsClient.find<RawAlert>({
+      filter: 'alert.attributes.executionStatus.status:error',
+      fields: ['executionStatus'],
+      type: 'alert',
+    });
+    return (
+      data.filter((item) => item.attributes.executionStatus.error?.reason === 'decrypt').length > 0
```

Comment:
This feel brittle.
Can we add some kind of helper function that ensures we use the same mechanism to identify the decryption errors as the we do to construct them?
It might mean you need to change the `AlertExecutionStatusErrorReasonValues` type fro man array to an enum, so you can do something like `AlertExecutionStatusErrorReasonValues.Decrypt` which makes sense to me 🤔

Reply:
Ya, I don't have a problem changing from the "set of strings" type to an enum.  It was feeling pretty good to me, but I was figuring we might find cases where an enum worked better.  Sigh.  :wave: Java!

Reply:
Make sense to me. I can do this in the current PR scope